### PR TITLE
add firstmedia.com

### LINKED ISSUE #2074: Add Indonesian EPG Source: firstmedia.com
### Source

https://www.firstmedia.com/tv-guide

### Notes

Hi, this is my first time in issues forum, and thank you for all your help.

I have been wondering if you can add the EPG source from Indonesian cable TV First Media.

If you can't, then can I have a personal tutorial on how to add site sources myself?

Thank you for your future response(s).

### PR DESCRIPTION
**Pull request by liimee**
some info:
- some channels doesn't have xmltv_id set yet--any help would be great
- I think it only has data for one day--correct me if I'm wrong
- might run into timeout--I think the site is to blame

closes #2074

### CODE REVIEW

File: sites/firstmedia.com/firstmedia.com.channels.xml [hunk 1]
```diff
@@ -0,0 +1,193 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<site site="firstmedia.com">
+    <channels>
+        <channel lang="id" xmltv_id="ABCAustralia.au" site_id="251">ABC Australia</channel>
```
Comment (freearhey) [inline sites/firstmedia.com/firstmedia.com.channels.xml:4]:
```suggestion
        <channel lang="en" xmltv_id="ABCAustralia.au" site_id="251">ABC Australia</channel>
```

Most channels have the guide in English.

https://github.com/iptv-org/epg/blob/master/CONTRIBUTING.md#how-do-i-add-a-program-guide-for-the-channel

File: sites/firstmedia.com/firstmedia.com.config.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+const dayjs = require("dayjs");
+const timezone = require("dayjs/plugin/timezone");
+const utc = require('dayjs/plugin/utc');
+
+dayjs.extend(timezone);
+dayjs.extend(utc);
+
+module.exports = {
+  site: "firstmedia.com",
+  days: 1,
```
Comment (freearhey) [inline sites/firstmedia.com/firstmedia.com.config.js:10]:
```suggestion
  days: 1,
  request: {
    timeout: 30000 // 30 seconds
  },
```

Prevents a timeout error:

```sh
[1/1] firstmedia.com - ABCAustralia.au - Jun 19, 2023 (0 programs)
  Error: Connection timeout
```

https://github.com/freearhey/epg-grabber#site-config

**Review by freearhey — CHANGES_REQUESTED** [review]

File: sites/firstmedia.com/firstmedia.com.config.js (hunk 2 above)
Comment (liimee) [reply sites/firstmedia.com/firstmedia.com.config.js:10]: hmm, that didn't work for me?
```
Starting...
Loading 'sites/firstmedia.com/firstmedia.com.config.js'...
Loading 'sites/firstmedia.com/firstmedia.com.channels.xml'..
[1/144] firstmedia.com - ABCAustralia.au - Jun 21, 2023 (0 programs)
  Error: connect ETIMEDOUT 2404:5600:0:3::2:443
[2/144] firstmedia.com - AlJazeeraChannel.qa - Jun 21, 2023 (0 programs)
  Error: connect ETIMEDOUT 2404:5600:0:3::2:443
```

Comment (freearhey) [reply sites/firstmedia.com/firstmedia.com.config.js:10]: I get the error all the time without a `timeout`:

```sh
npx epg-grabber --config=sites/firstmedia.com/firstmedia.com.config.js --channels=sites/firstmedia.com/firstmedia.com.channels.xml --output=guide.xml
Starting...
Loading 'sites/firstmedia.com/firstmedia.com.config.js'...
Loading 'sites/firstmedia.com/firstmedia.com.channels.xml'...
[1/144] firstmedia.com - ABCAustralia.au - Jun 21, 2023 (32 programs)
[2/144] firstmedia.com - AlJazeeraChannel.qa - Jun 21, 2023 (31 programs)
[3/144] firstmedia.com - AlQuranAlKareemTV.sa - Jun 21, 2023 (0 programs)
  Error: Connection timeout
```

and with it, everything works:

```sh
npx epg-grabber --config=sites/firstmedia.com/firstmedia.com.config.js --channels=sites/firstmedia.com/firstmedia.com.channels.xml --output=guide.xml
Starting...
Loading 'sites/firstmedia.com/firstmedia.com.config.js'...
Loading 'sites/firstmedia.com/firstmedia.com.channels.xml'...
[1/144] firstmedia.com - ABCAustralia.au - Jun 21, 2023 (32 programs)
[2/144] firstmedia.com - AlJazeeraChannel.qa - Jun 21, 2023 (31 programs)
[3/144] firstmedia.com - ZhejiangSatelliteTV.cn - Jun 21, 2023 (26 programs)
```

Comment (liimee) [reply sites/firstmedia.com/firstmedia.com.config.js:10]: still didn't work for me but I guess it's not related to the config I guess so alright then

**Review by BellezaEmporium — APPROVED** [review]

**Pull request merged**